### DISCUSSION TITLE
Tesseract openMp

### DISCUSSION THREAD

**Discussion by hiaryAbeer** [adaptech-cz/Tesseract4Android#73] [Q&A] [answered]
Hi all,
I'm using tesseract with flow of frames and i'm trying to get the best performance, but i noticed that doesn't work when i switch between standard and openMP variants and crashes happen when comment this condition:
if (!tessProcessing) {
Log.e(TAG, "recognizeImage: Processing is in progress");
return;}

so i would ask why these two variants gave me same results(duration calculation)? is there anything i missed or they're really same? also when i use two instances of tessApi processing time is increased while i guess to have better results?
how to get best processing time? and is recommended to use multithreading, if yes how without crashes?

**Comment by Robyer** [comment, marked as answer]
Hi, if you need to process the flow of images sequentially (in order), then you would have single Tesseract instance and in that case it would be faster to use OpenMP variant.

If you want to process multiple images at once, then use standard variant and create multiple Tesseract instances and process the images this way in separate threads.

But you mention the sample project crashes when you switched it to OpenMP variant? I'll check that.

**Reply by Robyer** [reply]
@hiaryAbeer Btw I don't have crash when using the openmp variant in the sample project. Is everything working for you now?
